cads_adaptors/adaptors/cds.py

```diff
@@ -43,6 +43,20 @@ def __init__(
         # List of steps to perform after retrieving the data
         self.post_process_steps: list[dict[str, Any]] = [{}]
 
+    def retrieve_list_of_results(self, request: Request) -> list[str]:
```

Comment:
IMO this should be an abstract method and all adaptors should implement it instead of `.retrieve`.
It's not ideal that some of the adaptors override `retrieve` and other implement `retrieve_list_of_results`.

Reply:
I agree, but I would like to do that step wise as I want to ensure all the developers are aware this new method, and that there are good examples for them to follow when they refactor their adaptor code.

If you agree to not include this extra work here, after the merge of this PR, I will create tickets for the adaptors which have not been updated in this PR.

Reply:
Sure, you can merge